### PR TITLE
Add .editorconfig file

### LINKED ISSUE #4097: Add .editorconfig file
Our style follows the example given on http://editorconfig.org/ for Python and YAML
difficulty/newcomer

### PR DESCRIPTION
**Pull request by aryanshar**
.editorconfig : Added required .editorconfig files
Closes #4097 

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by aryanshar** [comment]
Hey @IpshitaC .I finally changed the commit message. You may review it again !

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+indent_style = space
+indent_size = 2
+
+# Tab indentation
+[Makefile]
+indent_style = tab
+
+
+
+
```
Comment (palash25) [inline .editorconfig]: @aryanshar you have 5 newlines at the end of your file.

Comment (aryanshar) [reply .editorconfig]: @palash25 Ok so now i've fixed it. Please review it again!

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+# Set default charset
+[*.{py}]
```
Comment (adhikasp) [inline .editorconfig]: This could be merged with the next section.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+charset = utf-8
+
+# 4 space indentation
+[*.py]
+indent_style = space
+indent_size = 4
+
+# Matches the exact files either package.json or .travis.yml
+[{package.json, .travis.yml}]
```
Comment (adhikasp) [inline .editorconfig]: We have many yml files beside `.travis.yml`, maybe set it to *.yml?

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+insert_final_newline = true
+
+# Set default charset
+[*.{py}]
+charset = utf-8
+
+# 4 space indentation
+[*.py]
+indent_style = space
+indent_size = 4
```
Comment (adhikasp) [inline .editorconfig]: Add `trim_trailing_whitespace` and `max_line_length`

**Pull request closed without merging**